addons/mail/wizard/mail_wizard_invite.py

```diff
@@ -1,57 +1,39 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+from datetime import datetime
 from lxml import etree
 from lxml.html import builder as html
 from markupsafe import Markup
 
 from odoo import _, api, fields, models
 from odoo.exceptions import UserError
-from odoo.tools import is_html_empty
 
 
 class Invite(models.TransientModel):
     """ Wizard to invite partners (or channels) and make them followers. """
     _name = 'mail.wizard.invite'
     _description = 'Invite wizard'
 
-    @api.model
-    def default_get(self, fields):
-        result = super(Invite, self).default_get(fields)
-        if 'message' not in fields:
-            return result
-
-        user_name = self.env.user.display_name
-        model = result.get('res_model')
-        res_id = result.get('res_id')
-        if model and res_id:
-            document = self.env['ir.model']._get(model).display_name
-            title = self.env[model].browse(res_id).display_name
-            text = _(
-                '%(user_name)s invited you to follow %(document)s document: %(title)s',
-                user_name=user_name,
-                document=document,
-                title=title,
-            )
-        else:
-            text = _('%(user_name)s invited you to follow a new document.', user_name=user_name)
-
-        result['message'] = Markup("<div><p>%s</p><p>%s</p></div>") % (
-            _('Hello,'),
-            text,
-        )
-        return result
-
     res_model = fields.Char('Related Document Model', required=True, help='Model of the followed resource')
     res_id = fields.Integer('Related Document ID', help='Id of the followed resource')
     partner_ids = fields.Many2many('res.partner', string='Recipients')
-    message = fields.Html('Message')
-    notify = fields.Boolean('Send Notification', default=True)
+    message = fields.Html('')
 
     def add_followers(self):
-        if not self.env.user.email:
+        for wizard in self:
+            Model = self.env[wizard.res_model]
+            document = Model.browse(wizard.res_id)
+
+            # filter partner_ids to get the new followers, to avoid sending email to already following partners
+            new_partners = wizard.partner_ids - document.sudo().message_partner_ids
+            document.message_subscribe(partner_ids=new_partners.ids)
+        return {'type': 'ir.actions.act_window_close'}
+
+    def add_followers_notify(self):
+        user_from = self.env.user
```

Comment:
Not sure we gain something with this variable, would use self.env.user in code directly